### PR TITLE
Produce gap packets in connection

### PR DESCRIPTION
**Pull request by dejancrn**

### CODE REVIEW

**Review by NikolaiShipilov — APPROVED** [review]
Looks good; I noticed a few minor adjustments required in the Doxygen comments and code reuse.

**Review by denise-opendaq — APPROVED** [review]

File: core/opendaq/signal/src/connection_impl.cpp [hunk 1]
```diff
@@ -3,34 +3,62 @@
...
+                    packets.emplace_back(packetPtr);
+                    LOGP_T("Packet enqueued.")
+                });
+
+            port.notifyPacketEnqueued();
+            return OPENDAQ_SUCCESS;
+        });
 }
 
 ErrCode INTERFACE_FUNC ConnectionImpl::enqueueOnThisThread(IPacket* packet)
```
[the remark this answers is not in the document]
Comment (dejancrn) [reply core/opendaq/signal/src/connection_impl.cpp:64]: Ok

File: core/opendaq/signal/include/opendaq/event_packet_ids.h [hunk 2]
```diff
@@ -25,4 +25,10 @@ namespace event_packet_id
      *  - "DomainDataDescriptor" : IDataDescriptor
      */
     const std::string DATA_DESCRIPTOR_CHANGED = "DATA_DESCRIPTOR_CHANGED";
-}
+
+    /*!
+     * Parameter dictionary elements:
+     *  - "GapSize" : Int64       // in ticks
```
[the remark this answers is not in the document]
Comment (dejancrn) [reply core/opendaq/signal/include/opendaq/event_packet_ids.h:31]: Ok

**Review by JakaMohorkoDS — APPROVED** [review]

File: core/opendaq/signal/src/connection_impl.cpp [hunk 3]
```diff
@@ -192,6 +224,176 @@ const std::deque<PacketPtr>& ConnectionImpl::getPackets() const noexcept
...
+        domainSampleType = domainDescriptor.getSampleType();
+        if (domainSampleType == SampleType::Float64)
+            delta.valueDouble = rule.getParameters()["delta"];
+        else if (domainSampleType == SampleType::Int64 || domainSampleType == SampleType::UInt64)
+            delta.valueInt64_t = rule.getParameters()["delta"];
+        else
+        {
+            LOGP_T("Gap check not available, invalid domain sample type.")
+            gapCheckState = GapCheckState::not_available;
+            return;
```
[the remark this answers is not in the document]
Comment (dejancrn) [reply core/opendaq/signal/src/connection_impl.cpp:366]: for now I would say yes

**Pull request merged**